### ISSUE TITLE
How to succesfully train GA-RPN from scratch？

### ISSUE THREAD

**Issue by zhengye1995** [open-mmlab/mmdetection#2478]
I try to train GA-RPN from scratch based on the config file of GA-RPN and train from scartch, but the loss can not converge, so how to set config file to train GA-RPN from scratch.

**Comment by hellock** [comment]
You mean training on COCO without the ImageNet pretrained weights?

**Comment by zhengye1995** [comment]
Yes，I mean training without ImageNet pretrained， I set pretrained=None and frozen_stages=-1 to make all layers trainable. When training naive RPN, the model can converge. However, when I change to GA-RPN, it can not converge any more, any suggestions?

**Comment by myownskyW7** [comment]
We haven't tried to train GA-RPN from scratch before. My suggestion is that you may adopt sync-bn rather than bn to train the model.

**Comment by zhengye1995** [comment]
Thanks for your suggestion, I use gn when training from scratch, I will try sync-bn.

**Comment by myownskyW7** [comment]
GN should also work.

**Comment by zhengye1995** [comment]
> GN should also work.

I try GN, but it can not converge.

**Comment by myownskyW7** [comment]
How do you apply GN?

**Comment by zhengye1995** [comment]
> How do you apply GN?
like this:
norm_cfg = dict(type='GN', num_groups=32, requires_grad=True)

I set this cfg in backbone and neck .

The detail of config file：
# model settings
norm_cfg = dict(type='GN', num_groups=32, requires_grad=True)
model = dict(
    type='RPN',
    pretrained=None,
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=-1,
        style='pytorch',
        norm_cfg=norm_cfg),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        num_outs=5,
        norm_cfg=norm_cfg),
    rpn_head=dict(
        type='GARPNHead',
        in_channels=256,
        feat_channels=256,
        octave_base_scale=8,
        scales_per_octave=3,
        octave_ratios=[0.5, 1.0, 2.0],
        anchor_strides=[4, 8, 16, 32, 64],
        anchor_base_sizes=None,
        anchoring_means=[.0, .0, .0, .0],
        anchoring_stds=[0.07, 0.07, 0.14, 0.14],
        target_means=(.0, .0, .0, .0),
        target_stds=[0.07, 0.07, 0.11, 0.11],
        loc_filter_thr=0.01,
        loss_loc=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_shape=dict(type='BoundedIoULoss', beta=0.2, loss_weight=1.0),
        loss_cls=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0),
        loss_bbox=dict(type='SmoothL1Loss', beta=1.0, loss_weight=1.0)))

**Comment by myownskyW7** [comment]
You may also try a longer warmup scheduler or sync-bn.

**Comment by zhengye1995** [comment]
Thanks, I will try it.

**Comment by zhengye1995** [comment]
> You may also try a longer warmup scheduler or sync-bn.

Unfortunately, GN or sync-bn with longer warmup scheduler  all do not work ,  I don't know why,  I just want to see if GA-RPN is useful to training from scratch.

**Comment by MPolaris** [comment]
you may also set norm_eval=True

**Comment by zhengye1995** [comment]
> you may also set norm_eval=False

Thanks, I will try it.

**Comment by hellock** [comment]
Did you use the 6x schedule? You may refer to the Faster R-CNN from scratch config.
